adr/ADR-31.md

```diff
@@ -104,18 +104,46 @@ Using the `batch` and `max_bytes` keys one can request multiple messages in a si
 
 When requests are made against servers that do not support `batch` the first response will be received and nothing will follow. Old servers can be detected by the absence of the `Nats-Num-Pending` header in the first reply.
 
+#### Multiple Subject requests
+
+Multiple subjects can be requested in the same manner that a Batch can be requested. In this mode we support consistent point in time reads by allowing for a group of subjects to be read as they were at a point in time - assuming the stream holds enough historical data.
+
+To help inform proper use of this feature, vs just using a consumer, a multi subject request may only allow matching no more than 1024 subjects, any more will result in a `413` status reply.
+
+Using requests like `{"multi_last":["$KV.USERS.1234.>"]}` all the latest values for all subjects below that wildcard will be returned.
+
+Specific data for a user could be requested using `{"multi_last":["$KV.USERS.1234.name", "$KV.USERS.1234.address"]}` rather than all the user data you will now only get the 2 specific keys.
+
+To facilitate consistent multi key reads the `up_to_seq` and `up_to_time` keys can be added which will restrict the results up to a certain point in time.
+
+Imagine we have a new bucket and we did:
+
+```
+$ nats kv put 1234.name Bob                # message seq 1
+$ nats kv put 1234.surname Smith           # message seq 2
+$ nats kv put 1234.address 1 Main Street   # message seq 3
+$ nats kv put 1234.address 10 Oak Lane     # message seq 4 updates message 3
+```
+
+If we did a normal multi read using `{"multi_last":["$KV.USERS.1234.>"]}` we would get the address `10 Oak Lane` returned.  However, to get a record as it was at a certain point in the past we could supply the sequence or time, adding `"up_to_seq":3` to the request will return address `1 Main Street` along with the other data.
+
+A `batch` parameter can be added to restrict the result set to a certain size, otherwise the server will decide when to end the batch using the same `EOB` marker message seen in Batched Mode with the addition of the `Nats-UpTo-Sequence` header
+
+When the server cannot send any more data it will respond, like the above Batch, with a zero length payload message will be sent with the `Nats-Num-Pending` and `Nats-Last-Sequence` headers set that clients can use to determine if further batch calls are needed. It would also have the `Status` header set to `204` with the `Description` header being `EOB`. Additionally the `Nats-UpTo-Sequence` would be set that indicates the last matching message in the stream that matched criteria.  This number would be used in subsequent requests as the `up_to_seq` value to ensure batches of multi-gets are done around a consistent point in time.
```

Comment:
```suggestion
When the server cannot send any more data it will respond, like the above Batch, with a zero-length payload message including the `Nats-Num-Pending` and `Nats-Last-Sequence` headers enabling clients to determine if further batch calls are needed. In addition, it would also have the `Status` header set to `204` with the `Description` header being `EOB`. The `Nats-UpTo-Sequence` header will be set indicating the last message in the stream that matched criteria. This number would be used in subsequent requests as the `up_to_seq` value to ensure batches of multi-gets are done around a consistent point in time.
```